### PR TITLE
SearchLog: Do not store IP of logged in users, add retention policy

### PR DESCRIPTION
**Pull request by riking**
Also set a time-based retention policy on search logs, in addition to the size-based retention policy.

### CODE REVIEW

File: config/locales/server.en.yml [hunk 1]
```diff
@@ -1067,6 +1067,7 @@ en:
     search_recent_posts_size: "How many recent posts to keep in the index"
     log_search_queries: "Log search queries performed by users"
     search_query_log_max_size: "Maximum amount of search queries to keep"
+    search_query_log_max_retention_days: "Maximum amount of time to keep search queries, in days. Default is 3 years (1095 days)."
```
Comment (SamSaffron) [inline config/locales/server.en.yml]: default should be 6 months, I do not see why we need to hoard 3 years here.

**Comment by SamSaffron** [comment]
retention should be tested with freeze_time

File: config/locales/server.en.yml (hunk 1 above)
Comment (riking) [reply config/locales/server.en.yml]: The admin dashboard is trying to do year-over-year analytics on search queries; should search logs be limited to month/quarter scope?

**Comment by SamSaffron** [comment]
sure we can limit that, yearly trends of search are not really that
interesting imo

File: config/locales/server.en.yml (hunk 1 above)
Comment (SamSaffron) [reply config/locales/server.en.yml]: largest interval there is a year, so just keep 1 year max I guess.

**Comment by SamSaffron** [comment]
Pushing this out 2 weeks so it is post 2.0 release, can you finish by then? not super urgent

**Comment by SamSaffron** [comment]
merging but will fiddle with the dates a bit

**Pull request merged**